embark.el

```diff
@@ -466,55 +471,63 @@ There are three kinds:
 
 (autoload 'dired-get-filename "dired")
 
 (defun embark-target-file-at-point ()
   "Target file at point.
-This function mostly relies on `ffap-file-at-point', with two exceptions:
-
-1. In `dired-mode', it uses `dired-get-filename' instead.
-
-2. In `emacs-lisp-mode', it only calls `ffap-file-at-point' if
-   point is in a string or comment, or if it is on symbol
-   preceded by `require' or `use-package'."
-  (when-let ((file (cond
-                    ((derived-mode-p 'dired-mode)
-                     (dired-get-filename t 'no-error-if-not-filep))
-                    ((derived-mode-p 'emacs-lisp-mode)
-                     (when (or (nth 3 (syntax-ppss)) ; in a string
-                               (nth 4 (syntax-ppss)) ; or comment
-                               (save-excursion
-                                 (unless (looking-at "\\_<")
-                                   (forward-symbol -1))
-                                 (forward-symbol -1)
-                                 (looking-at "use-package\\|require")))
-                       (ffap-file-at-point)))
-                     (t (ffap-file-at-point)))))
-    (cons 'file (abbreviate-file-name file))))
+This function mostly relies on `ffap-file-at-point', with one exception:
+In `dired-mode', it uses `dired-get-filename' instead."
+  (if-let (file (and (derived-mode-p 'dired-mode)
+                     (dired-get-filename t 'no-error-if-not-filep)))
+      (save-excursion
+        (end-of-line)
+        `(file ,(abbreviate-file-name file)
+               ,(save-excursion
+                  (re-search-backward " " (line-beginning-position) 'noerror)
+                  (1+ (point)))
+               . ,(point)))
+    (when-let (file (ffap-file-at-point))
+      `(file ,(abbreviate-file-name file)
+             . ,(bounds-of-thing-at-point 'filename)))))
 
 (defun embark-target-bug-reference-at-point ()
   "Target a bug reference at point."
-  (when-let ((url (seq-some (lambda (o) (overlay-get o 'bug-reference-url))
-                            (overlays-at (point)))))
-    (cons 'url url)))
+  (when-let ((ov (seq-find (lambda (ov) (overlay-get ov 'bug-reference-url))
+                           (overlays-at (point)))))
+    `(url ,(overlay-get ov 'bug-reference-url)
+          ,(overlay-start ov) . ,(overlay-end ov))))
 
 (defun embark-target-url-at-point ()
   "Target the URL at point."
   (when-let ((url (ffap-url-at-point)))
-    (cons 'url url)))
+    `(url ,url . ,(bounds-of-thing-at-point 'url))))
 
+(declare-function widget-at "wid-edit")
 (defun embark-target-custom-variable-at-point ()
   "Target the variable corresponding to the customize widget at point."
   (when (derived-mode-p 'Custom-mode)
-    (when-let ((symbol (get-text-property (point) 'custom-data)))
-      (cons 'variable (symbol-name symbol)))))
+    (save-excursion
+      (beginning-of-line)
+      ;; TODO Please check, the old custom variable finder did not work for me.
```

Comment:
I think the old one may have needed Emacs 28.

Reply:
Ah okay. I didn't test on 28. Did you add this recently?

Reply:
Yes, pretty recently. I'll test yours on Emacs 28. If this version work on 27 and 28, it's definitely better.